src/main/java/org/elasticsearch/index/query/QueryBuilder.java

```diff
@@ -45,4 +46,11 @@
      * @throws IOException
      */
     Query toQuery(QueryParseContext parseContext) throws QueryParsingException, IOException;
+
+    /**
+     * Validate the query.
+     * @return {@code null} if query is valid, otherwise {@link ActionRequestValidationException} containing error messages,
+     * e.g. if fields that are needed to create the lucene query are missing.
+     */
+    ActionRequestValidationException validate();
```

Comment:
Changed the method signature to `QueryValidationException validate()` and introduced own exception class, similar to the existing one.